### PR TITLE
Fix dnf repo file names in Rocky 9

### PR DESCRIPTION
**Pull request by mnasiadka**
Rocky 9 uses /etc/yum.repos.d/rocky.repo and rocky-extras.repo and the current config creates duplicate entries.

### CODE REVIEW

File: etc/kayobe/dnf.yml [hunk 1]
```diff
@@ -112,19 +112,19 @@ dnf_custom_repos_rocky:
   appstream:
     baseurl: "{{ stackhpc_repo_rocky_appstream_url }}"
     description: "Rocky Linux $releasever - AppStream"
-    file: Rocky-AppStream
+    file: "{{ 'Rocky-AppStream' if os_distribution == '8' else 'rocky' }}"
```
Comment (Alex-Welsh) [inline etc/kayobe/dnf.yml]:
```suggestion
    file: "{{ 'Rocky-AppStream' if os_release == '8' else 'rocky' }}"
```

File: etc/kayobe/dnf.yml [hunk 2]
```diff
@@ -112,19 +112,19 @@ dnf_custom_repos_rocky:
...
     gpgkey: file:///etc/pki/rpm-gpg/RPM-GPG-KEY-rockyofficial
     gpgcheck: yes
   baseos:
     baseurl: "{{ stackhpc_repo_rocky_baseos_url }}"
     description: "Rocky Linux $releasever - BaseOS"
-    file: Rocky-BaseOS
+    file: "{{ 'Rocky-BaseOS' if os_distribution == '8' else 'rocky' }}"
```
Comment (Alex-Welsh) [inline etc/kayobe/dnf.yml]:
```suggestion
    file: "{{ 'Rocky-BaseOS' if os_release == '8' else 'rocky' }}"
```

File: etc/kayobe/dnf.yml [hunk 3]
```diff
@@ -112,19 +112,19 @@ dnf_custom_repos_rocky:
...
     gpgkey: file:///etc/pki/rpm-gpg/RPM-GPG-KEY-rockyofficial
     gpgcheck: yes
   extras:
     baseurl: "{{ stackhpc_repo_rocky_extras_url }}"
     description: "Rocky Linux $releasever - Extras"
-    file: Rocky-Extras
+    file: "{{ 'Rocky-Extras' if os_distribution == '8' else 'rocky-extras' }}"
```
Comment (Alex-Welsh) [inline etc/kayobe/dnf.yml]:
```suggestion
    file: "{{ 'Rocky-Extras' if os_release == '8' else 'rocky-extras' }}"
```

**Review by Alex-Welsh — CHANGES_REQUESTED** [review]

File: etc/kayobe/dnf.yml (hunk 1 above)
Comment (mnasiadka) [reply etc/kayobe/dnf.yml]: Done

File: etc/kayobe/dnf.yml (hunk 2 above)
Comment (mnasiadka) [reply etc/kayobe/dnf.yml]: Done

File: etc/kayobe/dnf.yml (hunk 3 above)
Comment (mnasiadka) [reply etc/kayobe/dnf.yml]: Done

**Review by Alex-Welsh — APPROVED** [review]

**Pull request merged**